packages/flutter/lib/src/widgets/routes.dart

```diff
@@ -967,3 +967,85 @@ abstract class PopupRoute<T> extends ModalRoute<T> {
     super.didChangeNext(nextRoute);
   }
 }
+
+/// A [Navigator] observer that notifies [RouteAware]s of changes to the
+/// state of their [Route].
+///
+/// To make a [StatefulWidget] aware of its current [Route] state, implement
```

Comment:
This kind of code should be in a `## Sample Code` block so that it gets analyzed.